crates/solver/src/driver.rs

```diff
@@ -740,22 +743,33 @@ impl Driver {
                     .map(|(solver, settlement, _)| (solver, settlement))
                     .collect(),
             );
-            self.send_solver_competition(solver_competition).await;
+            self.send_solver_competition(next_solver_competition, solver_competition)
+                .await;
         }
         // Happens after settlement submission so that we do not delay it.
         self.report_simulation_errors(errors, current_block_during_liquidity_fetch, gas_price);
         Ok(())
     }
 
-    fn next_auction_id(&mut self) -> u64 {
-        let id = self.solve_id;
-        self.solve_id += 1;
+    fn next_run_id(&mut self) -> u64 {
+        let id = self.run_id;
+        self.run_id += 1;
         id
     }
 
-    async fn send_solver_competition(&self, body: SolverCompetition) {
+    async fn send_solver_competition(
+        &self,
+        expected_id: SolverCompetitionId,
+        body: SolverCompetition,
+    ) {
         match self.api.send_solver_competition(&body).await {
-            Ok(id) => tracing::info!(?id, "stored solver competition"),
+            Ok(id) if id == expected_id => tracing::info!("stored solver competition"),
+            Ok(actual_id) => {
+                tracing::warn!(
+                    %expected_id, %actual_id,
+                    "stored solver competition with unexpected ID",
+                );
```

Comment:
If we go down the road of letting the driver know the `expected_id` beforehand we should probably require it to send us this `expected_id` with its solution and move this check into the `autopilot`.
The solution wouldn't make sense for a different competition anyway so not storing it would should be no problem IMO.

Reply:
That makes sense to me. I would do this in another PR though.